Source/WebCore/dom/StaticRange.cpp

```diff
@@ -83,4 +83,23 @@ void StaticRange::visitNodesConcurrently(JSC::AbstractSlotVisitor& visitor) cons
     addWebCoreOpaqueRoot(visitor, end.container.get());
 }
 
+bool StaticRange::isValid() const
+{
+    Node& startContainer = this->startContainer();
+    Node& endContainer = this->endContainer();
+
+    if (collapsed())
+        return false;
+    if (endOffset() < startOffset() && startContainer.isSameNode(&endContainer))
+        return false;
+    if (startContainer.isDocumentNode() || startContainer.isDocumentTypeNode())
+        return false;
+    if (endContainer.isDocumentNode() || endContainer.isDocumentTypeNode())
+        return false;
+    if (!connectedInSameTreeScope(&startContainer.rootNode(), &endContainer.rootNode()))
+        return false;
+    if (is_gt(treeOrder<ComposedTree>(startContainer, endContainer)))
```

Comment:
This is a O(n) operation. We may want to consider calling this function computeValidity instead to signify the expensive nature of this function call.